src/pudl/metadata/classes.py

```diff
@@ -303,53 +212,71 @@ def _validator(*names, fn: Callable) -> Callable:
 
     Args:
         names: Names of attributes to validate.
-        fn: Validation function (see :meth:`pydantic.validator`).
+        fn: Validation function (see :meth:`pydantic.field_validator`).
 
     Examples:
-        >>> class Class(Base):
+        >>> class Class(BaseModel):
         ...     x: list = None
         ...     _check_unique = _validator("x", fn=_check_unique)
-        >>> Class(y=[0, 0])
+        >>> Class(x=[0, 0])
         Traceback (most recent call last):
         ValidationError: ...
     """
-    return pydantic.validator(*names, allow_reuse=True)(fn)
+    return field_validator(*names)(fn)
 
 
-# ---- Classes: Field ---- #
+########################################################################################
+# PUDL Metadata Classes
+########################################################################################
+class PudlMeta(BaseModel):
+    """A base model that configures some options for PUDL metadata classes."""
+
+    model_config = ConfigDict(
+        extra="forbid",
+        validate_all=True,
+        validate_assignment=True,
+    )
```

Comment:
This class is a simplified generic version of the old `Base` class, which sets some model configurations shared across all of our metadata classes below.